content/en/agent/faq/linux-key-rotation-2024.md

```diff
@@ -0,0 +1,201 @@
+
+## What happens if the new key is not trusted before it is rotated?
+
+Trying to install or upgrade Agent packages using `apt`, `yum`, `dnf` or `zypper` from `apt.datadoghq.com`/`yum.datadoghq.com` without trusting the new key results in an error.
+
+Possible errors include:
+
+```
+E: The repository 'https://apt.datadoghq.com stable Release' is not signed.
+```
+```
+E: Package 'datadog-agent' has no installation candidate
+```
+```
+The following signatures couldn't be verified because the public key is not available: NO_PUBKEY
+```
+```
+The GPG keys listed for the "Datadog, Inc." repository are already installed but they are not correct for this package.
+Check that the correct key URLs are configured for this repository.
+```
+```
+Public key for datadog-agent-7.57.1-1.x86_64.rpm is not installed. Failing package is: datadog-agent-1:7.57.1-1.x86_64
+```
+```
+Error: GPG check FAILED
+```
+
+For `apt`, this applies to both newly released and existing versions of the Agent. For `yum`, `dnf` or `zypper`, existing versions of the Agent can still be installed as long as `repo_gpgcheck=0` is set in the `datadog.repo` file.
+
+This key rotation does not affect installations done by manually downloading the packages and installing them with `dpkg` or `rpm`. This may cause a warning for `rpm`.
+
+## Manual update
+
+Datadog encourages you to use one of the [install methods](#install-methods-that-automatically-trust-the-new-gpg-key) above, which trust the new GPG key as well as all future keys automatically. If this is not an option, use the following instructions to manually download and trust the new key.
+
+{{< tabs >}}
+{{% tab "Debian/Ubuntu" %}}
+
+Run the following commands on the host:
+
+```bash
+$ curl -o /tmp/DATADOG_APT_KEY_C0962C7D https://keys.datadoghq.com/DATADOG_APT_KEY_C0962C7D.public
+$ sudo apt-key add /tmp/DATADOG_APT_KEY_C0962C7D
+$ sudo touch /usr/share/keyrings/datadog-archive-keyring.gpg
+$ cat /tmp/DATADOG_APT_KEY_C0962C7D | sudo gpg --import --batch --no-default-keyring --keyring /usr/share/keyrings/datadog-archive-keyring.gpg
+$ sudo chmod a+r /usr/share/keyrings/datadog-archive-keyring.gpg
+```
+
+{{% /tab %}}
+{{% tab "RedHat/CentOS/SUSE" %}}
+
+Run the following commands on the host:
+
+```
+$ curl -o /tmp/DATADOG_RPM_KEY_B01082D3 https://keys.datadoghq.com/DATADOG_RPM_KEY_B01082D3.public
+$ sudo rpm --import /tmp/DATADOG_RPM_KEY_B01082D3
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Check if a host trusts the new GPG key
+
+{{< tabs >}}
+{{% tab "Debian/Ubuntu" %}}
+
+A host correctly trusts the new key if either one of these conditions is true:
+
+- The file `/usr/share/keyrings/datadog-archive-keyring.gpg` exists and the Datadog source list file (usually `/etc/apt/sources.list.d/datadog.list`) contains the option `[signed-by=/usr/share/keyrings/datadog-archive-keyring.gpg]`.
```

Comment:
This applies to the Agent repo; we should also provide a similar point for OPW and `/etc/apt/sources.list.d/datadog-observability-pipelines-worker.list`

Reply:
I rephrased and grouped the Agent and OPW with "subconditions", let me know if you agree